pkg/controller/rds/dbinstance/setup.go

```diff
@@ -0,0 +1,190 @@
+package dbinstance
+
+import (
+	"context"
+
+	svcsdk "github.com/aws/aws-sdk-go/service/rds"
+	svcsdkapi "github.com/aws/aws-sdk-go/service/rds/rdsiface"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/password"
+	"github.com/crossplane/crossplane-runtime/pkg/ratelimiter"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	svcapitypes "github.com/crossplane/provider-aws/apis/rds/v1alpha1"
+	aws "github.com/crossplane/provider-aws/pkg/clients"
+	"github.com/crossplane/provider-aws/pkg/clients/rds"
+)
+
+const (
+	errGetSecretFailed    = "failed to get Kubernetes secret"
+	errUpdateSecretFailed = "failed to update Kubernetes secret"
+	errSaveSecretFailed   = "failed to save generated password to Kubernetes secret"
+)
+
+// SetupDBInstance adds a controller that reconciles DBInstance
+func SetupDBInstance(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter) error {
+	name := managed.ControllerName(svcapitypes.DBInstanceGroupKind)
+	opts := []option{
+		func(e *external) {
+			c := &custom{client: e.client, kube: e.kube}
+			e.preObserve = preObserve
+			e.postObserve = postObserve
+			e.preCreate = c.preCreate
+			e.postCreate = c.postCreate
+			e.preDelete = preDelete
+			e.filterList = filterList
```

Comment:
Could you use existing [LateInitialize](https://github.com/crossplane/provider-aws/blob/8aa68a2/pkg/clients/rds/rds.go#L368) and [IsUpToDate](https://github.com/crossplane/provider-aws/blob/8aa68a2/pkg/clients/rds/rds.go#L456) as well so that update works and we get feature-parity with existing `RDSInstance`?